src/EditProfile.js

```diff
@@ -0,0 +1,174 @@
+    }
+    componentDidMount(){
+        const myRequest = new Request('http://127.0.0.1:5000/viewprofile', {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({"userId": this.state.userId}),
+            });
+            fetch(myRequest)
+                .then(response => response.json())
+                .then(res => 
+                        this.setState({
+                            name: res.name,
+                            age: res.age, 
+                            bio: res.bio,
+                            gender: res.gender,
+                            genderPreference: res.genderPreference,
+                            education: res.education, 
+                            interests: res.interests,
+                            maxDistance: res.maxDistance,
+                        })
+            ).catch((error) => {
+                console.error(error)
+        })
+      }
+
+    handleUpdate = (event) => {
+      event.preventDefault();
+      if(this.completedInput()) {
+        const id = this.props?.location?.state?.id;
+        const myForm = new FormData (document.getElementById("profileForm"));
+        myForm.append("id", id);
+        const myRequest = new Request("http://127.0.0.1:5000/editprofile", {
+          method: "POST",
+          body: myForm,
+        });
+
+        fetch(myRequest)
+        .then((res) =>
+          res.json())
+        .then((res) => {
+          if(res.response === "Success"){
+            this.setState({
+                updateStatus: "Profile Just Updated!"
+              });
+          }
+            
+          else {
+            this.setState({
+              error: res.response,
+            });
+          }  
+        })
+        .catch((error) => {
+          this.setState({
+            error: "Error connecting to backend",
+          });
+        });
+      }
+      else 
+        alert("Please fill in all fields");
+    };
+
+    completedInput = () => {
+      const inputs = ['name', 'age', 'bio', 'gender', 'genderPreference', 'education', 'interests'];
+      for(var i =0; i <inputs.length; i++) {
+        if(!this.state[inputs[i]])
+          return false;
+      }
+      return true;
+    }
+
+    handleChange = (event) => {
+      const target = event.target;
+      const value = target.value;
+      const name = target.name;
+
+      this.setState({
+        [name]: value,
+      });
+    }
+
+    render() {
+      return (
+          
+        <div className="Profile">
+            <Link to={{pathname: '/main', state: {id: this.state.userId}}}>Back to Swiping!</Link>
+          <form id="profileForm" onSubmit={this.handleUpdate}>
+            <h1>My Profile</h1>
+
+            <p>Name:</p>
+
+            <input type="text" name="name" value = {this.state.name} onChange={this.handleChange} maxlength="30" />
+
+            <p>Age:</p>
+
+            <input
```

Comment:
this will probably need to be changed to work with the new date of birth implementation

Reply:
It will be changed with the new birthday setting.